domain/model/state/state.go

```diff
@@ -486,9 +486,84 @@ SELECT type FROM model_type;
 	})
 }
 
-// List returns a list of all model UUIDs in the system that have not been
+// ListAllModels returns a slice of all models in the controller. If no models
+// exist an empty slice is returned.
+func (s *State) ListAllModels(ctx context.Context) ([]coremodel.Model, error) {
+	db, err := s.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	modelStmt := `
+SELECT md.model_uuid, md.name, cl.name, cr.name, mt.type, u.uuid, u.name, cc.name, o.name, ccn.name, l.value
+FROM model_metadata AS md
+LEFT JOIN model_list ml ON ml.uuid = md.model_uuid
+LEFT JOIN cloud cl ON cl.uuid = md.cloud_uuid
+LEFT JOIN cloud_region cr ON cr.uuid = md.cloud_region_uuid
+LEFT JOIN cloud_credential cc ON cc.uuid = md.cloud_credential_uuid
+LEFT JOIN model_type mt ON mt.id = md.model_type_id
+LEFT JOIN user u ON u.uuid = md.owner_uuid
+LEFT JOIN user o ON o.uuid = cc.owner_uuid
```

Comment:
Maybe alias this as "cco"

Reply:
This will get updated in this PR when the DDL changes land. There is a new view that makes this all go away.